packages/core/src/dispatcher.ts

```diff
@@ -134,7 +141,21 @@ export default class Dispatcher extends EventEmitter {
    * @param cid - Record CID
    */
   async retrieveRecord (cid: CID | string): Promise<any> {
-    return cloneDeep((await this._ipfs.dag.get(cid)).value)
+    const record = await this._ipfs.dag.get(cid, { timeout: DEFAULT_GET_TIMEOUT })
+    await this._restrictRecordSize(cid)
```

Comment:
Would be nice to do this before loading the record, so we don't bother to load a very large record we're just going to throw out.

Reply:
I don't think it's possible to check size without getting the data from the network.

Reply:
Yeah, unless we want to throw a zero-knowledge proof into the IPFS implementation 😛

Reply:
ah, I saw that 'stat' method on IPFS and thought that it was a query to the IPFS network for someone who owns the contents to return the 'stat' info.  But I guess it's actually asking the local IPFS cache for 'stat' info on an already-loaded CID?

EDIT: Nothing in the [docs](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/BLOCK.md#ipfsblockstatcid-options) suggest that you have to load the document before you can access it via the `ipfs.block` api... Is that just a case of the IPFS docs being bad?

Reply:
I just assumed that this was the case. Not completely certain though.

Reply:
maybe worth a manual test to see what the behavior is @simonovic86?

Reply:
yeah, it makes sense to do that. I created a [620](https://github.com/ceramicnetwork/js-ceramic/issues/620) story for it.